src/Components/Checkbox.tsx

```diff
@@ -1,71 +1,114 @@
-import React from "react"
-
+import React, { Component, HTMLProps } from "react"
 import styled from "styled-components"
 import colors from "../Assets/Colors"
 
-interface Props extends React.HTMLProps<Checkbox> {
+export interface CheckboxState {
   checked: boolean
 }
 
-export class Checkbox extends React.Component<Props, null> {
-  static defaultProps = {
-    checked: true,
+export class Checkbox extends Component<HTMLProps<Checkbox>, CheckboxState> {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      checked: props.checked || false,
+    }
+  }
+
+  onChange = event => {
+    if (this.props.onChange) {
+      this.props.onChange(event)
+    }
+
+    this.setState({
+      checked: event.currentTarget.checked,
+    })
   }
 
   render() {
-    const { checked } = this.props
+    const { children, className, ...propsForCheckbox } = this.props
 
     return (
-      <div className={this.props.className}>
+      <Label className={className}>
         <CheckboxInput
-          onChange={() => null}
           type="checkbox"
-          checked={checked}
+          {...propsForCheckbox as any}
+          onChange={this.onChange}
+          checked={this.state.checked}
         />
-        <Label />
-      </div>
+
+        {children}
+      </Label>
     )
   }
 }
 
-const StyledCheckbox = styled(Checkbox)`
+const CheckboxInput = styled.input`
+  position: relative;
+  top: -1px;
   width: 20px;
   height: 20px;
-  position: relative;
-  user-select: none;
-  background: ${colors.grayMedium};
-  display: inline-block;
-`
+  margin: 0 0.5rem 0 0;
 
-const CheckboxInput = styled.input`
-  visibility: hidden;
-  &:checked + label:after {
-    opacity: 1 !important;
+  // The before represents the check mark
+  &:before {
+    transform: rotate(-45deg);
+    content: "";
+    position: absolute;
+    z-index: 1;
+    top: 5px;
+    left: 4px;
+    width: 0.6rem;
+    height: 0.3rem;
+    border: 2px solid rgba(0, 0, 0, 1);
+    border-top-style: none;
+    border-right-style: none;
+    transition: opacity 0.25s;
+    opacity: 0;
   }
-`
 
-const Label = styled.label`
-  cursor: pointer
-  position: absolute
-  top: 2px;
-  right: 2px;
-  bottom: 2px;
-  left: 2px;
-  background-color: white;
+  &:hover:before {
+    opacity: 0.1;
+  }
+
+  &:checked:before {
+    opacity: 1;
+  }
+
+  // The after represents the square box
   &:after {
-    content: '';
+    content: "";
     position: absolute;
-    top: 3px;
-    right: 2px;
-    bottom: 7px;
-    left: 2px;
-    border: 2px solid black;
-    border-top: none;
-    border-right: none;
-    opacity: 0;
-    transform: rotate(-45deg) translateZ(0);
-    transition: opacity 0.25s;
+    left: 0;
+    width: 1rem;
+    height: 1rem;
+    background: #fff;
```

Comment:
All these hex definitions could be updated to use `colors._` like you do in the line below

Reply:
:+1: I think I should replace all `color` occurrences with a `colors._`